src/components/Header.js

```diff
@@ -1,8 +1,27 @@
 import React from 'react';
 import logo from '../assets/logo.png';
 import { Link } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { auth } from '../firebase/firebase.config';
+import { removeUser } from '../redux/coderSlice';
+import { useDispatch } from 'react-redux';
+import {
+    signOut
+} from 'firebase/auth'
```

Comment:
Assuming the `firebase` here is the 3rd party library where as the `firstbase` importing the `auth` is a local directory. The naming is the same so it's a bit confusing. Can we rename the `firebase` folder in your local?